cmake/options.cmake

```diff
@@ -72,6 +72,10 @@ else()
     set(CMAKE_DISABLE_PRECOMPILE_HEADERS On CACHE BOOL "Disable precompiled headers")
 endif()
 
+if (CMAKE_VERSION VERSION_LESS "3.9")
+    message(FATAL_ERROR "Registering tests with CTest requires at least CMake 3.9. Use REGISTER_TESTS=OFF to disable this.")
```

Comment:
This would have bailed out in the test code later on before. I moved it up so there is not a stray `cmake_minimum_required()` call somewhere in the code.